### PR TITLE
set 'basic' tiername when creating the MUR

### PR DESCRIPTION
**Pull request by xcoulon**
this will be used by the member operator to
provision the namespaces afterwards

Fixes CRT-321

E2E tests PR - https://github.com/codeready-toolchain/toolchain-e2e/pull/30

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by xcoulon** [comment]
> Looks good. But also waiting for the e2e tests :)

e2e tests are finally available in https://github.com/codeready-toolchain/toolchain-e2e/pull/30

File: pkg/controller/usersignup/usersignup_controller.go [hunk 1]
```diff
@@ -180,9 +180,18 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
...
 		}
-
+		// look-up the `basic` NSTemplateTier to get the NS templates
+		var nstemplateTier toolchainv1alpha1.NSTemplateTier
+		err := r.client.Get(context.TODO(), types.NamespacedName{
+			Namespace: request.Namespace, // assume that NSTemplateTier were created in the same NS as Usersignups
+			Name:      "basic",
+		}, &nstemplateTier)
+		if err != nil {
+			reqLogger.Error(err, "No 'basic' NSTemplateTier found.")
```
Comment (alexeykazakov) [inline pkg/controller/usersignup/usersignup_controller.go]: Let's update status here.

Comment (xcoulon) [reply pkg/controller/usersignup/usersignup_controller.go]: done in https://github.com/codeready-toolchain/host-operator/pull/94/commits/bc9973a563e64570c9cbe8ed77fccdf2fa6e99ed

File: pkg/controller/usersignup/usersignup_controller.go [hunk 2]
```diff
@@ -180,9 +181,19 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
...
+			// let's requeue until the NSTemplateTier resource is available
+			return reconcile.Result{Requeue: true}, r.updateStatus(reqLogger, instance, r.setStatusNoTemplateTierAvailable)
```
Comment (alexeykazakov) [inline pkg/controller/usersignup/usersignup_controller.go]: Use `wrapErrorWithStatusUpdate()` instead. So we have the original error in the status too. See  https://github.com/codeready-toolchain/host-operator/blob/b2a7ccf79f1baff97da785689f2480bbfff65621/pkg/controller/usersignup/usersignup_controller.go#L147 for example.

File: pkg/controller/usersignup/usersignup_controller.go (hunk 1 above)
Comment (alexeykazakov) [reply pkg/controller/usersignup/usersignup_controller.go]: See https://github.com/codeready-toolchain/host-operator/pull/94#discussion_r345906061

File: pkg/controller/usersignup/usersignup_controller.go (hunk 2 above)
Comment (xcoulon) [reply pkg/controller/usersignup/usersignup_controller.go]: done in f29652e

**Comment by alexeykazakov** [comment]
There is a panic:
```
goroutine 16 [running]:
testing.tRunner.func1(0xc000176f00)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x146d4a0, 0x22c1c10)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/codeready-toolchain/toolchain-e2e/test/e2e.verifyResources(0xc000176f00, 0xc0008b5900, 0x161f67b, 0x9, 0xc000b65d30, 0x3, 0x3)
	/tmp/toolchain-e2e/test/e2e/e2e_test.go:331 +0x497
github.com/codeready-toolchain/toolchain-e2e/test/e2e.TestE2EFlow(0xc000176f00)
	/tmp/toolchain-e2e/test/e2e/e2e_test.go:37 +0x502
testing.tRunner(0xc000176f00, 0x16d3528)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350
```

Not sure why though...

**Comment by xcoulon** [comment]
> There is a panic:
> 
> ```
> goroutine 16 [running]:
> testing.tRunner.func1(0xc000176f00)
> 	/usr/local/go/src/testing/testing.go:874 +0x3a3
> panic(0x146d4a0, 0x22c1c10)
> 	/usr/local/go/src/runtime/panic.go:679 +0x1b2
> github.com/codeready-toolchain/toolchain-e2e/test/e2e.verifyResources(0xc000176f00, 0xc0008b5900, 0x161f67b, 0x9, 0xc000b65d30, 0x3, 0x3)
> 	/tmp/toolchain-e2e/test/e2e/e2e_test.go:331 +0x497
> github.com/codeready-toolchain/toolchain-e2e/test/e2e.TestE2EFlow(0xc000176f00)
> 	/tmp/toolchain-e2e/test/e2e/e2e_test.go:37 +0x502
> testing.tRunner(0xc000176f00, 0x16d3528)
> 	/usr/local/go/src/testing/testing.go:909 +0xc9
> created by testing.(*T).Run
> 	/usr/local/go/src/testing/testing.go:960 +0x350
> ```
> 
> Not sure why though...

there's a timeout while waiting for a condition, but I suspect that we call the "parent" `t` which usually causes such a panic...

```
e2e_test.go:326: 
        	Error Trace:	e2e_test.go:326
        	            				e2e_test.go:37
        	Error:      	Received unexpected error:
        	            	timed out waiting for the condition
        	Test:       	TestE2EFlow
```

**Comment by alexeykazakov** [comment]
> there's a timeout while waiting for a condition, but I suspect that we call the "parent" t which usually causes such a panic..

Hm... I thought they’ve fixed that issue with parent T in sub tests...

**Comment by xcoulon** [comment]
> > there's a timeout while waiting for a condition, but I suspect that we call the "parent" t which usually causes such a panic..
> 
> Hm... I thought they’ve fixed that issue with parent T in sub tests...

yes, that's right. And I may have spoken too fast, I'm still trying to understand why there's such a panic (and why there's a timeout, too 😬)

**Comment by MatousJobanek** [comment]
guys, the problem is not with the parent `T` but with the `userAccount` that wasn't retreived. See the panic line: 
```
github.com/codeready-toolchain/toolchain-e2e/test/e2e.verifyResources(0xc000176f00, 0xc0008b5900, 0x161f67b, 0x9, 0xc000b65d30, 0x3, 0x3)
	/tmp/toolchain-e2e/test/e2e/e2e_test.go:331 +0x497
```
which is exactly this: https://github.com/codeready-toolchain/toolchain-e2e/blob/afc6e486f38653d45008ce69dd0eb26bc409b751/test/e2e/e2e_test.go#L331 - the `userAccount` is nill because the wait timeouted here https://github.com/codeready-toolchain/toolchain-e2e/blob/afc6e486f38653d45008ce69dd0eb26bc409b751/test/e2e/e2e_test.go#L325 but as we use `assert.NoError` and not `require.NoError` then we proceed with `userAccount` having nil

**Comment by alexeykazakov** [comment]
Good catch Matous.

**Pull request merged**